In [117]:
import json
import urllib
import pandas as pd
import numpy as np
import requests
import json
import re
import matplotlib.pyplot as plt
from bs4 import BeautifulSoup
import random
import time
%matplotlib inline

We import our scraped dataset from the API lab, which has information on the movies in IMDB's top 250 movies list.

In [178]:
df = pd.read_csv('scrapetest.csv')
df.head(1)

Unnamed: 0                                             Actors  \
0           0  Tim Robbins, Morgan Freeman, Bob Gunton, Willi...   

                                              Awards Country        Director  \
0  Nominated for 7 Oscars. Another 19 wins & 30 n...     USA  Frank Darabont   

          Genre Language  Metascore  \
0  Crime, Drama  English       80.0   

                                                Plot  \
0  Two imprisoned men bond over a number of years...   

                                              Poster    ...         Released  \
0  https://images-na.ssl-images-amazon.com/images...    ...      14 Oct 1994   

  Response  Runtime                     Title   Type  \
0     True  142 min  The Shawshank Redemption  movie   

                                              Writer  Year     imdbID  \
0  Stephen King (short story "Rita Hayworth and S...  1994  tt0111161   

  imdbRating  imdbVotes  
0        9.3  1,786,262  

[1 rows x 21 columns]

We will add a new column that indicates that all these 250 movies are part of the list.

In [179]:
df['top250'] = 1

In [180]:
df.head(1)

Unnamed: 0                                             Actors  \
0           0  Tim Robbins, Morgan Freeman, Bob Gunton, Willi...   

                                              Awards Country        Director  \
0  Nominated for 7 Oscars. Another 19 wins & 30 n...     USA  Frank Darabont   

          Genre Language  Metascore  \
0  Crime, Drama  English       80.0   

                                                Plot  \
0  Two imprisoned men bond over a number of years...   

                                              Poster   ...   Response  \
0  https://images-na.ssl-images-amazon.com/images...   ...       True   

   Runtime                     Title   Type  \
0  142 min  The Shawshank Redemption  movie   

                                              Writer  Year     imdbID  \
0  Stephen King (short story "Rita Hayworth and S...  1994  tt0111161   

   imdbRating  imdbVotes  top250  
0         9.3  1,786,262       1  

[1 rows x 22 columns]

In [181]:
df.drop('Unnamed: 0', axis = 1, inplace = True)

We make a list of the imdbIDs so we can check if any of our new scraped data is already in the top 250 list.

In [6]:
id250 = list(df['imdbID'])

We will scrape data from various IMDB pages to acquire data from movies that aren't in the top 250 list.

We will aim for at least 250 new movies

In [25]:
#checker
checker = 250

Lets get some movies that are currently in theaters

In [7]:
top_movies = requests.get('http://www.imdb.com/chart/boxoffice?ref_=nv_ch_cht_1')

with open('top_boxoffice.html', 'w') as f:
    f.write(top_movies.content)



In [10]:
with open('top_boxoffice.html', 'r') as f:
    content = BeautifulSoup(f.read())

In [16]:
main_table = content.find('tbody') #<tdbody>

In [17]:
movie_ids = []

for movie in main_table.find_all('td', {'class': 'watchlistColumn'}):
    for div_tag in movie.find_all('div'):
        try:
            movie_ids.append(div_tag['data-tconst'])
        except:
            pass

In [19]:
movie_rows = []
for movie in movie_ids:
    r = requests.get('http://www.omdbapi.com/?i=%s&r=json' % movie)
    json_content = json.loads(r.content)
    movie_rows.append(json_content)

In [39]:
dfbox = pd.DataFrame(movie_rows)
dfbox.shape
dfbox.to_csv('dfbox.csv', encoding = 'utf-8')
checker = checker - dfbox.shape[0]

In [27]:
checker

240

240 more movies to go, lets randomly sample movies by year. We want a diverse range of movies, so we will scrape by genres. However, IMDB only sorts movies by genre by popularity. We will scrape as many movies as possible and hopefully have a large enough sample size to randomly select movies to fill up our dataset.

Here we scrape 100 of the worst movies on IMDB

In [40]:
bot_movies = requests.get('http://www.imdb.com/chart/bottom')

with open('bottom.html', 'w') as f:
    f.write(bot_movies.content)


In [41]:
with open('bottom.html', 'r') as f:
    content = BeautifulSoup(f.read())

In [42]:
main_table = content.find('tbody', {'class': 'lister-list'}) #<tbody class='lister-list'>

In [44]:
movie_ids = []

for movie in main_table.find_all('td', {'class': 'ratingColumn'}):
    for div_tag in movie.find_all('div'):
        try:
            movie_ids.append(div_tag['data-titleid'])
        except:
            pass

In [45]:
movie_rows = []
for movie in movie_ids:
    r = requests.get('http://www.omdbapi.com/?i=%s&r=json' % movie)
    json_content = json.loads(r.text)
    movie_rows.append(json_content)

In [47]:
dfbot = pd.DataFrame(movie_rows)
dfbot.to_csv('dfbot.csv', encoding = 'utf-8')

Since I need as many movies as possible, to save time, I downloaded a dataset of 5000 IMDB movies scraped around 7 months ago that was uploaded to Kaggle. 

In [49]:
df5000 = pd.read_csv('movie_metadata.csv')
df5000.head()

color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4    NaN        Doug Walker                     NaN       NaN   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    131.0                     NaN        Rob Walker   

   actor_1_facebook_likes        gross                           genres  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi   
1                 40000.0  309404152.0         Action|Adventure|Fantasy   
2                 11000.0  200074175.0        Action|Adventure|Thriller   
3                 27000.0  448130642.0                  Action|Thriller   
4                   131.0          NaN                      Documentary   

          ...          num_user_for_reviews language  country  content_rating  \
0         ...                        3054.0  English      USA           PG-13   
1         ...                        1238.0  English      USA           PG-13   
2         ...                         994.0  English       UK           PG-13   
3         ...                        2701.0  English      USA           PG-13   
4         ...                           NaN      NaN      NaN             NaN   

        budget  title_year actor_2_facebook_likes imdb_score  aspect_ratio  \
0  237000000.0      2009.0                  936.0        7.9          1.78   
1  300000000.0      2007.0                 5000.0        7.1          2.35   
2  245000000.0      2015.0                  393.0        6.8          2.35   
3  250000000.0      2012.0                23000.0        8.5          2.35   
4          NaN         NaN                   12.0        7.1           NaN   

  movie_facebook_likes  
0                33000  
1                    0  
2                85000  
3               164000  
4                    0  

[5 rows x 28 columns]

Since the dataset has features that are different than our original dataset, we only want to extract the imdbID, and feed it to OMDBAPI again. To do this, we have to split the movie_imdb_link column values.

In [55]:
df5000['movie_imdb_link'][0].split('/')[4]
#this gives us the imdbID

'tt0499549'

In [88]:
lis5000 = [x.split('/')[4] for x in df5000['movie_imdb_link']]

However, it might be better to add the IMDBid's of our previous scraped datasets as well and sample from one list

In [68]:
box = [str(x) for x in dfbox['imdbID']]
bot = [str(x) for x in dfbot['imdbID']]

In [90]:
lis5000.extend(box)

In [91]:
lis5000.extend(bot)

In [92]:
len(lis5000)

5153

We don't want to imbalance our dataset with 250 1s and 5153 0s, so we will only sample 250 movies from the list. 

In [114]:
samplelis = [x for x in lis5000 if x not in id250]
len(samplelis)

4988

In [115]:
sampleid = random.sample(samplelis, 250)

In [142]:
movie_rows = []
for movie in sampleid[0:200]:
    r = requests.get('http://www.omdbapi.com/?i=%s&r=json' % movie)
    json_content = json.loads(r.content)
    movie_rows.append(json_content)
    time.sleep(1)

In [143]:
dfsample1 = pd.DataFrame(movie_rows)
dfsample1.to_csv('dfsample1.csv', encoding = 'utf-8')

In [145]:
movie_rows = []
for movie in sampleid[200:251]:
    r = requests.get('http://www.omdbapi.com/?i=%s&r=json' % movie)
    json_content = json.loads(r.content)
    movie_rows.append(json_content)
    time.sleep(2)

In [146]:
dfsample2 = pd.DataFrame(movie_rows)
dfsample2.to_csv('dfsample2.csv', encoding = 'utf-8')

I just had to separate the movie IDs as I was rate-limited when I tried requesting all 250 at once.

In [147]:
listofdf = [dfsample1, dfsample2]
dftomerge = pd.concat(listofdf)
dftomerge.shape

(250, 21)

In [148]:
dftomerge.reset_index(drop = True)

Actors  \
0    Adam Beach, Evan Adams, Irene Bedard, Gary Farmer   
1    Robin Williams, Mandy Moore, John Krasinski, E...   
2    Victor Rasuk, Donna Maldonado, Kevin Rivera, K...   
3    Russell Crowe, Elizabeth Banks, Michael Buie, ...   
4    Dane Cook, Kate Hudson, Alec Baldwin, Jason Biggs   
5    Mike Starr, Paul Herman, Don Pardo, Martin Ros...   
6    Nicole Kidman, Joseph Fiennes, Hugo Weaving, L...   
7    Jordana Brewster, Taylor Handley, Diora Baird,...   
8    Siem Vroom, Marlies van Alcmaer, Erik van 't W...   
9    Madeline Carroll, Callan McAuliffe, Rebecca De...   
10   Shanley Caswell, Maureen McCormick, Eric Rober...   
11   Lily Collins, Jamie Campbell Bower, Kevin Zege...   
12   Denzel Washington, Angelina Jolie, Queen Latif...   
13   Shirley MacLaine, Ricki Lake, Brendan Fraser, ...   
14   Mickey Rourke, Don Johnson, Chelsea Field, Dan...   
15   Mark Wahlberg, Jennifer Aniston, Dominic West,...   
16         Jackie Chan, Lung Ti, Anita Mui, Felix Wong   
17   Hal Scardino, Litefoot, Lindsay Crouse, Richar...   
18   Justin Timberlake, Ted Danson, Harry Dean Stan...   
19   Nicholas Farrell, Nigel Havers, Ian Charleson,...   
20   Kevin Hart, Taraji P. Henson, Will 'Spank' Hor...   
21   Will Smith, Margot Robbie, Adrian Martinez, Ge...   
22   Kit Harington, Carrie-Anne Moss, Emily Brownin...   
23   Sylvester Stallone, Billy Dee Williams, Lindsa...   
24   Whoopi Goldberg, Gérard Depardieu, Haley Joel ...   
25   Woody Harrelson, Elias Koteas, Michael Kelly, ...   
26   Richard Dreyfuss, Ving Rhames, Elisabeth Shue,...   
27   Melissa McCarthy, Kristen Bell, Peter Dinklage...   
28   Andy Samberg, Jorma Taccone, Bill Hader, Danny...   
29   Christian Slater, Patricia Arquette, Dennis Ho...   
..                                                 ...   
220  Anthony Hopkins, Colin O'Donoghue, Alice Braga...   
221  Lindsay Lohan, Chris Pine, Samaire Armstrong, ...   
222  Robert De Niro, Anne Hathaway, Rene Russo, And...   
223  Samuel L. Jackson, Kurt Russell, Jennifer Jaso...   
224  James Nesbitt, Genevieve O'Reilly, Patrick O'K...   
225  Chris Bisson, Saeed Jaffrey, Sally Bankes, Pet...   
226  Mark Wahlberg, Rachel Weisz, Susan Sarandon, S...   
227  Jack Lemmon, Walter Matthau, Susan Sarandon, V...   
228  Leslie Nielsen, Nicollette Sheridan, Charles D...   
229  Billy Roche, Eanna Hardwicke, Hannah Lynch, Ci...   
230  Tom Cruise, Morgan Freeman, Olga Kurylenko, An...   
231  James Spader, Angela Bassett, Robert Forster, ...   
232  Michael J. Fox, Geena Davis, Hugh Laurie, Jona...   
233  Amber Heard, Anson Mount, Michael Welch, Whitn...   
234  Diego Luna, Zoe Saldana, Channing Tatum, Ron P...   
235  Jason Lee, Anjul Nigam, Brighton Sharbino, Hil...   
236                        Vera Ngassa, Beatrice Ntuba   
237  Neil Maskell, MyAnna Buring, Harry Simpson, Mi...   
238  Barry Bostwick, Michael Beck, Persis Khambatta...   
239  Giuseppe Andrews, James DeBello, Edward Furlon...   
240  Miguel A. Núñez Jr., Vivica A. Fox, Kevin Poll...   
241  Ace Marrero, Katie Stegeman, Jack E. Curenton,...   
242  Wesley Snipes, Woody Harrelson, Jennifer Lopez...   
243  Bruce Campbell, Grace Thorsen, Taylor Sharpe, ...   
244  Suraj Sharma, Irrfan Khan, Ayush Tandon, Gauta...   
245  Mary Alice, Tanveer K. Atwal, Helmut Bakaitis,...   
246  Anton Yelchin, Colin Farrell, Toni Collette, D...   
247  John Simm, Lorraine Pilkington, Shaun Parkes, ...   
248  Ross Noble, Tommy Knight, Shane Murray-Corcora...   
249  Konstantin Khabenskiy, Vladimir Menshov, Valer...   

                                                Awards  \
0                             11 wins & 8 nominations.   
1                                        1 nomination.   
2                             4 wins & 12 nominations.   
3                                       4 nominations.   
4                                        1 nomination.   
5    Nominated for 2 Oscars. Another 3 wins & 7 nom...   
6                               1 win & 6 nom

We see that we have included some TV series as well. We need to remove these and scrape additional samples

In [150]:
dftomerge['Type'].value_counts()

movie     243
series      7
Name: Type, dtype: int64

In [153]:
dftomerge = dftomerge.loc[dftomerge['Type'] == 'movie']
dftomerge.shape

(243, 21)

In [154]:
moresample = random.sample(samplelis, 7)

In [158]:
dftomergelis = [str(x) for x in dftomerge['imdbID']]

In [160]:
movie_rows = []
for movie in moresample:
    r = requests.get('http://www.omdbapi.com/?i=%s&r=json' % movie)
    json_content = json.loads(r.content)
    movie_rows.append(json_content)
    time.sleep(2)

In [161]:
dfsample3 = pd.DataFrame(movie_rows)
dfsample3.to_csv('dfsample3.csv', encoding = 'utf-8')

In [162]:
dfsample3

Actors  \
0  Zac Efron, Wes Bentley, Emily Ratajkowski, Jon...   
1  James Woods, Sonja Smits, Debbie Harry, Peter ...   
2  Robert Downey Jr., Gwyneth Paltrow, Don Cheadl...   
3  Elizabeth Berkley, Kyle MacLachlan, Gina Gersh...   
4  Kirk Cameron, Darren Doane, Bridgette Cameron,...   
5  Anika Noni Rose, Bruno Campos, Keith David, Mi...   
6  Frank Grillo, Carmen Ejogo, Zach Gilford, Kiel...   

                                              Awards          Country  \
0                                                N/A  UK, France, USA   
1                            3 wins & 7 nominations.           Canada   
2  Nominated for 1 Oscar. Another 17 wins & 55 no...       China, USA   
3                          10 wins & 10 nominations.      France, USA   
4                            4 wins & 2 nominations.              USA   
5  Nominated for 3 Oscars. Another 12 wins & 39 n...              USA   
6                            2 wins & 6 nominations.      France, USA   

                    Director                         Genre         Language  \
0                 Max Joseph         Drama, Music, Romance          English   
1           David Cronenberg      Horror, Sci-Fi, Thriller          English   
2                Shane Black     Action, Adventure, Sci-Fi          English   
3             Paul Verhoeven                         Drama          English   
4               Darren Doane                Comedy, Family          English   
5  Ron Clements, John Musker  Animation, Adventure, Comedy  English, French   
6             James DeMonaco        Action, Horror, Sci-Fi          English   

  Metascore                                               Plot  \
0        46  Caught between a forbidden romance and the exp...   
1        60  A sleazy cable-TV programmer begins to see his...   
2        62  When Tony Stark's world is torn apart by a for...   
3        16  Nomi, a young drifter, arrives in Las Vegas to...   
4        18  Kirk is enjoying the annual Christmas party ex...   
5        73  A waitress, desperate to fulfill her dreams as...   
6        50  Three groups of people intertwine and are left...   

                                              Poster  Rated     Released  \
0  https://images-na.ssl-images-amazon.com/images...      R  28 Aug 2015   
1  https://images-na.ssl-images-amazon.com/images...      R  04 Feb 1983   
2  https://images-na.ssl-images-amazon.com/images...  PG-13  03 May 2013   
3  https://images-na.ssl-images-amazon.com/images...  NC-17  22 Sep 1995   
4  https://images-na.ssl-images-amazon.com/images...     PG  14 Nov 2014   
5  https://images-na.ssl-images-amazon.com/images...      G  11 Dec 2009   
6  https://images-na.ssl-images-amazon.com/images...      R  18 Jul 2014   

  Response  Runtime                      Title   Type  \
0     True   96 min        We Are Your Friends  movie   
1     True   87 min                 Videodrome  movie   
2     True  130 min                 Iron Man 3  movie   
3     True  128 min                  Showgirls  movie   
4     True   80 min           Saving Christmas  movie   
5     True   97 min  The Princess and the Frog  movie   
6     True  103 min         The Purge: Anarchy  movie   

                                              Writer  Year     imdbID  \
0  Max Joseph (screenplay), Meaghan Oppenheimer (...  2015  tt3787590   
1                                   David Cronenberg  1983  tt0086541   
2  Drew Pearce (screenplay), Shane Black (screenp...  2013  tt1300854   
3                                      Joe Eszterhas  1995  tt0114436   
4                       Darren Doane, Cheston Hervey  2014  tt4009460   
5  Ron Clements (story by), John Musker (story by...  2009  tt0780521   
6  James DeMonaco, James DeMonaco (characters cre...  2014  tt2975578   

  imdbRating imdbVotes  
0        6.1    24,225  
1        7.3    63,782  
2        7.2   583,743  
3        4.6    51,685  
4        1.6    12,497  
5        7.1    93,672  
6        6.5   104,795

In [163]:
listofdf = [dftomerge, dfsample3]
dftomerge = pd.concat(listofdf)
dftomerge.reset_index(drop = True)

Actors  \
0    Adam Beach, Evan Adams, Irene Bedard, Gary Farmer   
1    Robin Williams, Mandy Moore, John Krasinski, E...   
2    Victor Rasuk, Donna Maldonado, Kevin Rivera, K...   
3    Russell Crowe, Elizabeth Banks, Michael Buie, ...   
4    Dane Cook, Kate Hudson, Alec Baldwin, Jason Biggs   
5    Mike Starr, Paul Herman, Don Pardo, Martin Ros...   
6    Nicole Kidman, Joseph Fiennes, Hugo Weaving, L...   
7    Jordana Brewster, Taylor Handley, Diora Baird,...   
8    Siem Vroom, Marlies van Alcmaer, Erik van 't W...   
9    Madeline Carroll, Callan McAuliffe, Rebecca De...   
10   Shanley Caswell, Maureen McCormick, Eric Rober...   
11   Lily Collins, Jamie Campbell Bower, Kevin Zege...   
12   Denzel Washington, Angelina Jolie, Queen Latif...   
13   Shirley MacLaine, Ricki Lake, Brendan Fraser, ...   
14   Mickey Rourke, Don Johnson, Chelsea Field, Dan...   
15   Mark Wahlberg, Jennifer Aniston, Dominic West,...   
16         Jackie Chan, Lung Ti, Anita Mui, Felix Wong   
17   Hal Scardino, Litefoot, Lindsay Crouse, Richar...   
18   Justin Timberlake, Ted Danson, Harry Dean Stan...   
19   Nicholas Farrell, Nigel Havers, Ian Charleson,...   
20   Kevin Hart, Taraji P. Henson, Will 'Spank' Hor...   
21   Will Smith, Margot Robbie, Adrian Martinez, Ge...   
22   Kit Harington, Carrie-Anne Moss, Emily Brownin...   
23   Sylvester Stallone, Billy Dee Williams, Lindsa...   
24   Whoopi Goldberg, Gérard Depardieu, Haley Joel ...   
25   Woody Harrelson, Elias Koteas, Michael Kelly, ...   
26   Richard Dreyfuss, Ving Rhames, Elisabeth Shue,...   
27   Melissa McCarthy, Kristen Bell, Peter Dinklage...   
28   Andy Samberg, Jorma Taccone, Bill Hader, Danny...   
29   Christian Slater, Patricia Arquette, Dennis Ho...   
..                                                 ...   
220  Jack Lemmon, Walter Matthau, Susan Sarandon, V...   
221  Leslie Nielsen, Nicollette Sheridan, Charles D...   
222  Billy Roche, Eanna Hardwicke, Hannah Lynch, Ci...   
223  Tom Cruise, Morgan Freeman, Olga Kurylenko, An...   
224  James Spader, Angela Bassett, Robert Forster, ...   
225  Michael J. Fox, Geena Davis, Hugh Laurie, Jona...   
226  Amber Heard, Anson Mount, Michael Welch, Whitn...   
227  Diego Luna, Zoe Saldana, Channing Tatum, Ron P...   
228  Jason Lee, Anjul Nigam, Brighton Sharbino, Hil...   
229                        Vera Ngassa, Beatrice Ntuba   
230  Neil Maskell, MyAnna Buring, Harry Simpson, Mi...   
231  Barry Bostwick, Michael Beck, Persis Khambatta...   
232  Giuseppe Andrews, James DeBello, Edward Furlon...   
233  Miguel A. Núñez Jr., Vivica A. Fox, Kevin Poll...   
234  Ace Marrero, Katie Stegeman, Jack E. Curenton,...   
235  Wesley Snipes, Woody Harrelson, Jennifer Lopez...   
236  Bruce Campbell, Grace Thorsen, Taylor Sharpe, ...   
237  Suraj Sharma, Irrfan Khan, Ayush Tandon, Gauta...   
238  Mary Alice, Tanveer K. Atwal, Helmut Bakaitis,...   
239  Anton Yelchin, Colin Farrell, Toni Collette, D...   
240  John Simm, Lorraine Pilkington, Shaun Parkes, ...   
241  Ross Noble, Tommy Knight, Shane Murray-Corcora...   
242  Konstantin Khabenskiy, Vladimir Menshov, Valer...   
243  Zac Efron, Wes Bentley, Emily Ratajkowski, Jon...   
244  James Woods, Sonja Smits, Debbie Harry, Peter ...   
245  Robert Downey Jr., Gwyneth Paltrow, Don Cheadl...   
246  Elizabeth Berkley, Kyle MacLachlan, Gina Gersh...   
247  Kirk Cameron, Darren Doane, Bridgette Cameron,...   
248  Anika Noni Rose, Bruno Campos, Keith David, Mi...   
249  Frank Grillo, Carmen Ejogo, Zach Gilford, Kiel...   

                                                Awards  \
0                             11 wins & 8 nominations.   
1                                        1 nomination.   
2                             4 wins & 12 nominations.   
3                                       4 nominations.   
4                                        1 nomination.   
5    Nominated for 2 Oscars. Another 3 wins & 7 nom...   
6                               1 win & 6 nom

In [167]:
dftomerge.drop('totalSeasons', axis = 1, inplace = True)

In [169]:
dftomerge['top250'] = 0

In [183]:
df.columns

Index([u'Actors', u'Awards', u'Country', u'Director', u'Genre', u'Language',
       u'Metascore', u'Plot', u'Poster', u'Rated', u'Released', u'Response',
       u'Runtime', u'Title', u'Type', u'Writer', u'Year', u'imdbID',
       u'imdbRating', u'imdbVotes', u'top250'],
      dtype='object')

In [184]:
dftomerge.columns

Index([    u'Actors',     u'Awards',    u'Country',   u'Director',
            u'Genre',   u'Language',  u'Metascore',       u'Plot',
           u'Poster',      u'Rated',   u'Released',   u'Response',
          u'Runtime',      u'Title',       u'Type',     u'Writer',
             u'Year',     u'imdbID', u'imdbRating',  u'imdbVotes',
           u'top250'],
      dtype='object')

In [185]:
dffinal = pd.concat([df,dftomerge])
dffinal.shape

(500, 21)

In [188]:
dffinal.reset_index(drop = True)

Actors  \
0    Tim Robbins, Morgan Freeman, Bob Gunton, Willi...   
1    Marlon Brando, Al Pacino, James Caan, Richard ...   
2    Al Pacino, Robert Duvall, Diane Keaton, Robert...   
3    Christian Bale, Heath Ledger, Aaron Eckhart, M...   
4    Martin Balsam, John Fiedler, Lee J. Cobb, E.G....   
5    Liam Neeson, Ben Kingsley, Ralph Fiennes, Caro...   
6    Tim Roth, Amanda Plummer, Laura Lovelace, John...   
7     Noel Appleby, Ali Astin, Sean Astin, David Aston   
8    Eli Wallach, Clint Eastwood, Lee Van Cleef, Al...   
9    Edward Norton, Brad Pitt, Meat Loaf, Zach Grenier   
10   Alan Howard, Noel Appleby, Sean Astin, Sala Baker   
11   Mark Hamill, Harrison Ford, Carrie Fisher, Bil...   
12   Tom Hanks, Rebecca Williams, Sally Field, Mich...   
13   Leonardo DiCaprio, Joseph Gordon-Levitt, Ellen...   
14   Bruce Allpress, Sean Astin, John Bach, Sala Baker   
15   Michael Berryman, Peter Brocco, Dean R. Brooks...   
16   Robert De Niro, Ray Liotta, Joe Pesci, Lorrain...   
17   Keanu Reeves, Laurence Fishburne, Carrie-Anne ...   
18   Toshirô Mifune, Takashi Shimura, Keiko Tsushim...   
19   Mark Hamill, Harrison Ford, Carrie Fisher, Pet...   
20   Alexandre Rodrigues, Leandro Firmino, Phellipe...   
21   Morgan Freeman, Andrew Kevin Walker, Kevin Spa...   
22   Jodie Foster, Lawrence A. Bonney, Kasi Lemmons...   
23   James Stewart, Donna Reed, Lionel Barrymore, T...   
24   Stephen Baldwin, Gabriel Byrne, Benicio Del To...   
25   Roberto Benigni, Nicoletta Braschi, Giorgio Ca...   
26   Jean Reno, Gary Oldman, Natalie Portman, Danny...   
27   Rumi Hiiragi, Miyu Irino, Mari Natsuki, Takash...   
28   Tom Hanks, Tom Sizemore, Edward Burns, Barry P...   
29   Claudia Cardinale, Henry Fonda, Jason Robards,...   
..                                                 ...   
470  Jack Lemmon, Walter Matthau, Susan Sarandon, V...   
471  Leslie Nielsen, Nicollette Sheridan, Charles D...   
472  Billy Roche, Eanna Hardwicke, Hannah Lynch, Ci...   
473  Tom Cruise, Morgan Freeman, Olga Kurylenko, An...   
474  James Spader, Angela Bassett, Robert Forster, ...   
475  Michael J. Fox, Geena Davis, Hugh Laurie, Jona...   
476  Amber Heard, Anson Mount, Michael Welch, Whitn...   
477  Diego Luna, Zoe Saldana, Channing Tatum, Ron P...   
478  Jason Lee, Anjul Nigam, Brighton Sharbino, Hil...   
479                        Vera Ngassa, Beatrice Ntuba   
480  Neil Maskell, MyAnna Buring, Harry Simpson, Mi...   
481  Barry Bostwick, Michael Beck, Persis Khambatta...   
482  Giuseppe Andrews, James DeBello, Edward Furlon...   
483  Miguel A. Núñez Jr., Vivica A. Fox, Kevin Poll...   
484  Ace Marrero, Katie Stegeman, Jack E. Curenton,...   
485  Wesley Snipes, Woody Harrelson, Jennifer Lopez...   
486  Bruce Campbell, Grace Thorsen, Taylor Sharpe, ...   
487  Suraj Sharma, Irrfan Khan, Ayush Tandon, Gauta...   
488  Mary Alice, Tanveer K. Atwal, Helmut Bakaitis,...   
489  Anton Yelchin, Colin Farrell, Toni Collette, D...   
490  John Simm, Lorraine Pilkington, Shaun Parkes, ...   
491  Ross Noble, Tommy Knight, Shane Murray-Corcora...   
492  Konstantin Khabenskiy, Vladimir Menshov, Valer...   
493  Zac Efron, Wes Bentley, Emily Ratajkowski, Jon...   
494  James Woods, Sonja Smits, Debbie Harry, Peter ...   
495  Robert Downey Jr., Gwyneth Paltrow, Don Cheadl...   
496  Elizabeth Berkley, Kyle MacLachlan, Gina Gersh...   
497  Kirk Cameron, Darren Doane, Bridgette Cameron,...   
498  Anika Noni Rose, Bruno Campos, Keith David, Mi...   
499  Frank Grillo, Carmen Ejogo, Zach Gilford, Kiel...   

                                                Awards  \
0    Nominated for 7 Oscars. Another 19 wins & 30 n...   
1      Won 3 Oscars. Another 23 wins & 27 nominations.   
2      Won 6 Oscars. Another 10 wins & 20 nominations.   
3    Won 2 Oscars. Another 146 wins & 142 nominations.   
4    Nominated for 3 Oscars. Another 16 wins & 8 no...   
5      Won 7 Oscars. Another 71 wins & 33 nominations.   
6       Won 1 Oscar. Another 60 wins & 65 nom

In [189]:
dffinal.to_csv('dffinal.csv', encoding = 'utf-8')